packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,513 @@
+      final TextStyle textStyle = DefaultTextStyle.of(context).style.copyWith(
+        color: (_selectedIndex == currentKey) ?
+        CupertinoColors.white : CupertinoColors.activeBlue,
+      );
+      final IconThemeData iconTheme = IconThemeData(
+        color: (_selectedIndex == currentKey) ?
+        CupertinoColors.white : CupertinoColors.activeBlue,
+      );
+
+      Widget child = widget.children[currentKey];
+      child = GestureDetector(
+        onTapDown: (TapDownDetails event) {
+          _onTapDown(currentKey);
+        },
+        onTapUp: _onTapUp,
+        onTapCancel: _onTapCancel,
+        onTap: () {
+          _onTap(currentKey);
+        },
+        child: new IconTheme(
+          data: iconTheme,
+          child: DefaultTextStyle(
+            style: textStyle,
+            overflow: TextOverflow.ellipsis,
+            child: Semantics(
+              inMutuallyExclusiveGroup: true,
+              selected: _selectedIndex != null ? _selectedIndex == currentKey : false,
+              child: child,
+            ),
+          ),
+        ),
+      );
+      _gestureChildren.add(child);
+    }
+
+    final Widget box = new _SegmentedControlRenderWidget<T>(
+      children: _gestureChildren,
+      selectedIndex: _selectedIndex,
+      pressedIndex: _pressedIndex,
+    );
+
+    return Padding(
+      padding: _kHorizontalItemPadding.resolve(Directionality.of(context)),
+      child: UnconstrainedBox(
+        constrainedAxis: Axis.horizontal,
+        child: box,
+      ),
+    );
+  }
+}
+
+class _SegmentedControlRenderWidget<T> extends MultiChildRenderObjectWidget {
+  _SegmentedControlRenderWidget({
+    Key key,
+    List<Widget> children = const <Widget>[],
+    @required this.selectedIndex,
+    @required this.pressedIndex,
+  }) : super(
+          key: key,
+          children: children,
+        );
+
+  final int selectedIndex;
+  final int pressedIndex;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return new _RenderSegmentedControl<T>(
+      textDirection: Directionality.of(context),
+      selectedIndex: selectedIndex,
+      pressedIndex: pressedIndex,
+    );
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderSegmentedControl<T> renderObject) {
+    renderObject
+      ..textDirection = Directionality.of(context)
+      ..selectedIndex = selectedIndex
+      ..pressedIndex = pressedIndex;
+  }
+}
+
+class _SegmentedControlContainerBoxParentData extends ContainerBoxParentData<RenderBox> {
+  RRect surroundingRect;
+}
+
+class _RenderSegmentedControl<T> extends RenderBox with
+    ContainerRenderObjectMixin<RenderBox, ContainerBoxParentData<RenderBox>>,
+    RenderBoxContainerDefaultsMixin<RenderBox, ContainerBoxParentData<RenderBox>>{
+  _RenderSegmentedControl({
+    List<RenderBox> children,
+    @required int selectedIndex,
+    @required int pressedIndex,
+    @required TextDirection textDirection,
+  })  : _textDirection = textDirection,
```

Comment:
Are there any asserts that are needed here?  Like maybe that some things are not null?  Just to be paranoid... :-)